prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py

```diff
@@ -0,0 +1,302 @@
+import sys
+import typing as t
+from datetime import datetime
+
+import subgrounds.subgraph
+from eth_typing import ChecksumAddress
+from hexbytes import HexBytes
+from subgrounds import FieldPath, Subgrounds
+
+from prediction_market_agent_tooling.gtypes import HexAddress
+from prediction_market_agent_tooling.markets.agent_market import FilterBy, SortBy
+from prediction_market_agent_tooling.markets.omen.data_models import (
+    OMEN_FALSE_OUTCOME,
+    OMEN_TRUE_OUTCOME,
+    OmenBet,
+    OmenMarket,
+    OmenUserPosition,
+    RealityAnswer,
+)
+from prediction_market_agent_tooling.tools.utils import to_int_timestamp, utcnow
+
+
+class OmenSubgraphHandler:
+    """
+    Class responsible for handling interactions with Omen subgraphs (trades, conditionalTokens).
+    """
+
+    OMEN_TRADES_SUBGRAPH = "https://api.thegraph.com/subgraphs/name/protofire/omen-xdai"
+    CONDITIONAL_TOKENS_SUBGRAPH = (
+        "https://api.thegraph.com/subgraphs/name/gnosis/conditional-tokens-gc"
+    )
+    REALITYETH_GRAPH_URL = (
+        "https://api.thegraph.com/subgraphs/name/realityeth/realityeth-gnosis"
+    )
+
+    # We define here as str for easier filtering.
+    INVALID_ANSWER_STR = HexBytes(
+        "0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
+    )
+
+    def __init__(self) -> None:
+        self.sg = Subgrounds()
+        # Load the subgraph
+        self.trades_subgraph = self.sg.load_subgraph(self.OMEN_TRADES_SUBGRAPH)
+        self.conditional_tokens_subgraph = self.sg.load_subgraph(
+            self.CONDITIONAL_TOKENS_SUBGRAPH
+        )
+        self.realityeth_subgraph = self.sg.load_subgraph(self.REALITYETH_GRAPH_URL)
+
+    def _get_fields_for_bets(self, bets_field: t.Any) -> list[FieldPath]:
```

Comment:
There are two ways to find out what type something should be:

With mypy, you can add `reveal_type(x)` to the code and run `mypy`, it will print it out.

However, that works only if the library is typed, if it isn't, you can do it in runtime, for example:

```python
print(
    type(OmenSubgraphHandler().trades_subgraph.Query.fixedProductMarketMaker(id="bla"))
)
```

which will print

```
<class 'subgrounds.subgraph.fieldpath.FieldPath'>
```

So

```suggestion
    def _get_fields_for_bets(self, bets_field: FieldPath) -> list[FieldPath]:
```

Reply:
Completely missed this type! Thanks, adding now